packages/actor-rdf-parse-html-script/lib/ActorRdfParseHtmlScript.ts

```diff
@@ -0,0 +1,99 @@
+import {
+  ActorRdfParseFixedMediaTypes,
+  IActionRdfParse,
+  IActionRootRdfParse,
+  IActorOutputRootRdfParse,
+  IActorRdfParseFixedMediaTypesArgs,
+  IActorRdfParseOutput,
+  IActorTestRootRdfParse,
+} from "@comunica/bus-rdf-parse";
+import {ActionContext, Actor, Mediator} from "@comunica/core";
+import {Readable} from "stream";
+
+/**
+ * A HTML script RDF Parse actor that listens on the 'rdf-parse' bus.
+ *
+ * It is able to extract and parse any RDF serialization from HTML files
+ * and announce the presence of them by media type.
+ */
+export class ActorRdfParseHtmlScript extends ActorRdfParseFixedMediaTypes {
+
+  public mediatorRdfParse: Mediator<Actor<IActionRootRdfParse, IActorTestRootRdfParse,
+    IActorOutputRootRdfParse>, IActionRootRdfParse, IActorTestRootRdfParse, IActorOutputRootRdfParse>;
+
+  constructor(args: IActorRdfParseFixedMediaTypesArgs) {
+    super(args);
+  }
+
+  public async runHandle(action: IActionRdfParse, mediaType: string, context: ActionContext):
+    Promise<IActorRdfParseOutput> {
+
+    const quads = new Readable({ objectMode: true } );
+
+    let initialized = false;
+
+    quads._read = async () => {
+      if (!initialized) {
+        initialized = true;
+
+        // Stringify HTML input
+        const htmlString: string = await require('stream-to-string')(action.input);
+
+        // Extract script tags
+        const DOMParser = require('xmldom').DOMParser;
+        const doc = new DOMParser().parseFromString(htmlString, 'text/html');
+        const scripts = doc.getElementsByTagName('script');
+
+        const supportedTypes: string[] = Object.keys((await this.mediatorRdfParse
+          .mediate({
+            context: ActionContext({ '@comunica/bus-rdf-parse:source': { type: 'mediaTypes' } }),
+            mediaTypes: true,
+          })).mediaTypes);
+
+        // Loop script elements
+        let count: number = 0;      // count streams
+        let amountOfWrongTypes: number = 0;
```

Comment:
I don't think these two separate counters are needed. Furthermore, the logic behind these counters seems incorrect.